### PR TITLE
fix: transaction version v2 does not apply

### PR DESCRIPTION
**Pull request by khanti42**
This PR is to change the management of transaction version. 

- The account instance will not set any transaction version. 
- Transaction version are set by the caller function. 
- Default transaction is 0x1.
- Available transaction version is 0x1, 0x3 (0x2 should not be used)

### CODE REVIEW

File: packages/starknet-snap/src/rpcs/executeTxn.ts [hunk 1]
```diff
@@ -117,6 +117,7 @@ export class ExecuteTxnRpc extends AccountRpcController<
         address,
         accountDeployed,
         calls,
+        suggestedMaxFee,
```
Comment (khanti42) [inline packages/starknet-snap/src/rpcs/executeTxn.ts:120]: this a small miss I found, not related to v1/v2 change but the UX only

File: packages/starknet-snap/src/utils/starknetUtils.ts [hunk 2]
```diff
@@ -1079,7 +1075,7 @@ export async function getEstimatedFees(
     suggestedMaxFee: estimateFeeResp.suggestedMaxFee.toString(10),
     overallFee: estimateFeeResp.overall_fee.toString(10),
     unit:
-      transactionVersion === constants.TRANSACTION_VERSION.V2
+      invocationsDetails.version === constants.TRANSACTION_VERSION.V1
```
Comment (stanleyyconsensys) [inline packages/starknet-snap/src/utils/starknetUtils.ts:1078]:
```suggestion
      invocationsDetails?.version === constants.TRANSACTION_VERSION.V3 ? FeeTokenUnit.STRK: FeeTokenUnit.ETH
```

it is better to if it is v3, then strk, everything else eth

File: packages/starknet-snap/src/utils/starknetUtils.ts [hunk 3]
```diff
@@ -1052,8 +1049,8 @@ export async function getEstimatedFees(
   transactionInvocations: Invocations,
   invocationsDetails: UniversalDetails = {},
 ): Promise<EstimateFeeResponse> {
-  const transactionVersion = (invocationsDetails?.version ??
-    TRANSACTION_VERSION) as TransactionVersion;
+  invocationsDetails.version =
+    invocationsDetails?.version ?? TRANSACTION_VERSION;
```
Comment (stanleyyconsensys) [inline packages/starknet-snap/src/utils/starknetUtils.ts:1053]:
```suggestion
```

File: packages/starknet-snap/src/utils/superstruct.ts [hunk 4]
```diff
@@ -109,11 +109,11 @@ export const NumberStringStruct = union([number(), HexStruct]);
 export const CairoVersionStruct = enums([CAIRO_VERSION, CAIRO_VERSION_LEGACY]);
 
 export const TxVersionStruct = enums([
-  constants.TRANSACTION_VERSION.V2,
+  constants.TRANSACTION_VERSION.V1,
```
Comment (stanleyyconsensys) [inline packages/starknet-snap/src/utils/superstruct.ts:112]: on signDeclareTransaction, it only accept v2 / v3 ^^

**Review by stanleyyconsensys — CHANGES_REQUESTED** [review]

**Pull request merged**